includes/core/server/request/class-paginated.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Paginated.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Server\Request;
+use WP_REST_Request;
+
+/**
+ * Request class for paginated requests.
+ */
+abstract class Paginated extends Request {
+
+
+	const DEFAULT_PARAMS = [
+		'page'      => 0,
+		'pagesize'  => 25,
+		'sort'      => 'created',
+		'direction' => 'desc',
+		'limit'     => 100,
+	];
+
+	const IMMUTABLE_PARAMS = [
+		'page',
+		'pagesize',
+		'sort',
+		'direction',
+		'limit',
+	];
+
+
+	/**
+	 * Returns the request's HTTP method.
+	 */
+	public function get_method(): string {
+		return 'GET';
+	}
+
+	/**
+	 * Used to prepare request from WP Rest data.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return Paginated
+	 */
+	public static function from_rest_request( $request ) {
+		$wcpay_request = static::create();
+		$wcpay_request->set_page( (int) $request->get_param( 'page' ) );
+		$wcpay_request->set_page_size( (int) ( $request->get_param( 'pagesize' ) ?? 25 ) );
+		$sort = $request->get_param( 'sort' );
+		if ( null !== $sort ) {
+			$wcpay_request->set_sort_by( (string) $sort );
+		}
+		$direction = $request->get_param( 'direction' );
+		if ( null !== $direction ) {
+			$wcpay_request->set_sort_direction( (string) $direction );
+		}
+		return $wcpay_request;
+	}
+
+	/**
+	 * Set filters.
+	 *
+	 * @param array $filters Filterd to set.
+	 *
+	 * @return void
+	 */
+	public function set_filters( array $filters ) {
+
+		foreach ( $filters as $key => $value ) {
+			// Nullable filters are not needed and mutable keys can only be changed with this method.
+			if ( null !== $value && ! in_array( $key, self::IMMUTABLE_PARAMS, true ) ) {
```

Comment:
Thanks for switching from default parameters to an immutable declaration. With that in mind though, I don't see any reason to keep the `is_array( ... )` check here. If it's needed for whatever reason, I don't see overwriting paging parameters as mission-critical, so let consumers overwrite them :)